src/test/searchAddressFromZipcode.test.ts

```diff
@@ -0,0 +1,105 @@
+import { expect } from '@jest/globals';
+import { addresses } from './constants/masterData/address';
+import { expectedadAddressDataListIncludHyphen } from './constants/masterData/address';
+import { expectedadAddressDataListExcludingHyphen } from './constants/masterData/address';
+import { fullinputZipcodeExcludingHyphen } from './constants/userInput/zipcode';
+
+test('ユーザー入力から取得した郵便番号の全角数字を半角に変換 ハイフンが入っていても数字のみの抽出', () => {
+  const testDataExcludingHyphen = '７３２００２１';
+  const testDataIncludingHyphen = '７３２ー００２１';
+  const testDatamixtureFullAndHalf = '７３2００２1';
+  const expectedDataExcludingHyphen = '7320021';
+
+  const convertZipCode = (testData: string): string => {
+    const a: string = testData.replace(/[０-９]/g, (s: string) =>
+      String.fromCharCode(s.charCodeAt(0) - 65248)
+    );
+    const b: RegExpMatchArray | [] = a.match(/\d/g) || [];
+    return b.join('');
+  };
+
+  expect(convertZipCode(testDataExcludingHyphen)).toBe(
+    expectedDataExcludingHyphen
+  );
+  expect(convertZipCode(testDataIncludingHyphen)).toBe(
+    expectedDataExcludingHyphen
+  );
+  expect(convertZipCode(testDatamixtureFullAndHalf)).toBe(
+    expectedDataExcludingHyphen
+  );
+});
+
+test('検索条件と郵便番号の桁数によって処理するか否かのフラグを返す', () => {
+  const isExactTrue = true;
+  const isExactFalse = false;
+  const lengthNotSeven = 8;
+  const lengthSeven = 7;
+  const lengthLessThanTwo = 1;
+  const lengthTwo = 2;
+  let flag = true;
+
+  const checkLength = (isExact: boolean, length: number) => {
+    // 完全一致検索の場合は入力データが7文字ちょうどで処理を実行
+    if (isExact === true && length !== 7) {
+      return (flag = false);
+    }
+
+    // 部分一致検索の場合は入力データが2文字以上で処理を実行
+    if (isExact === false && length <= 1) {
+      return (flag = false);
+    }
+
+    flag = true;
+
+    return flag;
+  };
+
+  // 完全一致検索で入力データが7文字なら処理が実行されるか
+  expect(checkLength(isExactTrue, lengthSeven)).toBe(true);
+
+  // 部分一致検索で入力データが2文字以上なら処理が実行されるか
+  expect(checkLength(isExactFalse, lengthTwo)).toBe(true);
+
+  // 完全一致検索の場合は7文字以外の場合は処理が実行されないか
+  expect(checkLength(isExactTrue, lengthNotSeven)).toBe(false);
+
+  // 部分一致検索の場合は2文字未満の場合は処理されないか
+  expect(checkLength(isExactFalse, lengthLessThanTwo)).toBe(false);
+});
+
+test('オプションでハイフンありを指定している場合、郵便番号にハイフンを追加する', () => {
+  const convertHyphenatedZipCode = (testOptionData: boolean) => {
```

Comment:
@marumoto-goq 
この引数名だと誤解を受けます。データ？オブジェクトかな？ってなってしまう。typescriptでboolean指定していますがなるべく変数名で書いた方が認知負担は減ります。

ここの場合だと以下でどうでしょうか。
```suggestion
  const convertHyphenatedZipCode = (hasOptionHyphen: boolean) => {
```

Reply:
@iwata-goq 
確かにそうですね、、、、
変数名しっかり見直して他変数名も見直しました！
[こちらのコミット](https://github.com/GoQSystemInc/GoqZipCode/pull/34/commits/5c570921cbb51e5b413cb55e4ab73bbde891b613)なので一度見ていただきたいです🙇‍♂️